drivers/video/gc2145.c

```diff
@@ -1039,7 +1039,7 @@ static int gc2145_set_fmt(const struct device *dev, enum video_endpoint_id ep,
 	}
 
 	/* Check if camera is capable of handling given format */
-	for (int i = 0; i == ARRAY_SIZE(fmts); i++) {
+	for (int i = 0; i < RESOLUTIONS_MAX; i++) {
```

Comment:
That is right! There is a `{0}` terminator! So the `ARRAY_SIZE()` is too large for it.
Since there is a `{0}` terminator, how about tis:
```suggestion
	for (int i = 0; i < fmts[i].pixelformat; i++) {
```
`.pixelformat` is always assured to be non-0 as it is a `four-cc` code.
I am also asking for putting a "reference implementation" for image sensor drivers, as these kind of challenges are repeated over and over for each and every sensor, and can be fixed once for all.

Thanks for pointing out!

Reply:
@facchinm ping?